src/services/tracing.ts

```diff
@@ -2,20 +2,96 @@ import {
   SemanticAttributes,
   SemanticResourceAttributes,
 } from "@opentelemetry/semantic-conventions"
+import type * as graphqlTypes from "graphql"
 import { W3CTraceContextPropagator } from "@opentelemetry/core"
 import { NodeTracerProvider } from "@opentelemetry/sdk-trace-node"
-import { getNodeAutoInstrumentations } from "@opentelemetry/auto-instrumentations-node"
+import { HttpInstrumentation } from "@opentelemetry/instrumentation-http"
+import { GraphQLInstrumentation } from "@opentelemetry/instrumentation-graphql"
+import { MongoDBInstrumentation } from "@opentelemetry/instrumentation-mongodb"
+import { IORedisInstrumentation } from "@opentelemetry/instrumentation-ioredis"
+import { GrpcInstrumentation } from "@opentelemetry/instrumentation-grpc"
 import { registerInstrumentations } from "@opentelemetry/instrumentation"
 import { SimpleSpanProcessor } from "@opentelemetry/sdk-trace-base"
 import { JaegerExporter } from "@opentelemetry/exporter-jaeger"
 import { Resource } from "@opentelemetry/resources"
-import { trace, context, propagation, SpanAttributes } from "@opentelemetry/api"
+import {
+  trace,
+  context,
+  propagation,
+  Span,
+  SpanAttributes,
+  SpanStatusCode,
+} from "@opentelemetry/api"
 import { tracingConfig } from "@config/app"
 
 propagation.setGlobalPropagator(new W3CTraceContextPropagator())
 
+const gqlResponseHook = (span: Span, data: graphqlTypes.ExecutionResult) => {
+  if (data.errors && data.errors.length > 0) {
```

Comment:
this could be squeezed to `data?.errors.length > 0` I think.

Reply:
`errors` is always an array?

Reply:
According to the type definition yes... there will be a follow up PR shortly. Will fix it.

Reply:
shouldn't it be `data.errors?.length`?